[16.0][IMP] spreadsheet_oca: Improve security

etobella opened the pull request:
With this change we are adding an extra layer of security. Without it, any user was able to sniff how all happened using something like:
```
const any_spreadsheet_id = 1234;
const channel = "spreadsheet_oca;spreadsheet.spreadsheet" + ";" + any_spreadsheet_id;
bus_service.addChannel(channel);
bus_service.addEventListener("spreadsheet_oca", (message) => /* every revision arrives here */
```

With the change, we verify the access to the model with a similar logic to `web_editor` fields

legalsylvain reviewed (commented):
Is it possible to remove read access to a spreadshéet for agroup and / or a user ? If yes, token should change at that moment.

etobella committed:
[FIX] spreadsheet_oca: Improve security

With this change we are adding an extra layer of security. Without it, any user was able to sniff how all happened using something like:

    const any_spreadsheet_id = 1234;
    const channel = "spreadsheet_oca;spreadsheet.spreadsheet;" + any_spreadsheet_id;
    bus_service.addChannel(channel);
    bus_service.addEventListener("spreadsheet_oca", (message) => /* every revision arrives here */

With the change, we verify the access to the model with a similar logic to `web_editor` fields

etobella commented:
I made a change in order to make it in a similar way to web_editor fields. Now it should be safe!! :smile: 

Any comments and reviews are welcome!!!

pedrobaeza reviewed (commented):
Can you put a link to the reference code and also a comment in the method docstring?

Isn't a try/catch block needed for not raising exception and simply don't add the channel to the list?

etobella commented:
Well, I wanted to avoid exceptions, the system just removes the channel if you don't have access. It seems clean enough to me :thinking:

pedrobaeza commented:
OK, and what about someone being removed from the access? Is it covered?

etobella commented:
Yes, each time it creates the bus, we check if the user has permissions. If we remove the permissions, the bus will remove the channel, even if you send it :smile:

pedrobaeza commented:
OK, then please squash to proceed with the merge.

etobella commented:
Rebase and message of the commit improved adding all the necessary information of the PR :smile:

pedrobaeza commented:
/ocabot merge patch
